reverse_engineering/queries/selectedTablesSubQuery/DatabaseIndexesSubQueryForRetrievingTheTablesSelectedByTheUser.js

```diff
@@ -0,0 +1,31 @@
+const { QueryForRetrievingTheTablesSelectedByTheUser } = require('./QueryForRetrievingTheTablesSelectedByTheUser');
+const { getProjectedPropertiesNames } = require('./getProjectedPropertiesNames');
+
+class DatabaseIndexesSubQueryForRetrievingTheTablesSelectedByTheUser {
```

Comment:
At first sight, this class looks 100% the same as `PartitionsSubQueryForRetrievingTheTablesSelectedByTheUser`. I suggest removing those 2 classes and just providing a default `projection` in the `QueryForRetrievingTheTablesSelectedByTheUser` (Sonar reported duplicated code as well)...